source/manuals/programming-languages/python/python.html.md.erb

```diff
@@ -41,6 +41,55 @@ _For more on linting (including standard settings and config) see the [linting][
 * Lint checks using [PyFlakes][]
 
 
+### Dependencies
+
+It is important to have a good strategy for specifying your project's Python
+libraries. There are two desirable characteristics:
+
+1. Reproducibility
+
+    Pin your full dependencies or you'll get unpredictability between your dev
+    environment and other environments. You want a new starter to avoid small
+    hard-to-spot problems. And you want parity between what you test locally,
+    what is tested by CI and what you deploy, or you risk new issues appearing
+    on a live server. Additionaly these things can be hard to diagnose.
+
+    Simply manually maintaining a requirements.txt doesn't capture
+    sub-dependencies. And we've noticed [minor problems with using
+    piptools/pip-compile and pipenv/Pipfile][dm-deps-commit].
+
+    **Recommendation**: Put your top-level requirements in a
+    `requirements-app.txt` file and use [this Makefile
```

Comment:
If anyone other than just Digital Marketplace are also using `requirements-dev.txt` then we could probably also mention that here too (i.e. dependencies for test/development not needed for production).

Reply:
It's such a common thing to do and it seems reasonable to standardize on the filename. requirements-dev.txt gets 10k results on Github, compared to 6k results for dev-requirements.txt, so I'll add your suggestion.